openmc/universe.py

```diff
@@ -402,22 +402,23 @@ def create_xml_subelement(self, xml_element):
         if len(self._name) > 0:
             element.set("name", str(self._name))
 
-        if isinstance(self._fill, openmc.Material):
-            element.set("material", str(self._fill._id))
+        if isinstance(self.fill, basestring):
+            assert self.fill.strip().lower() == 'void'
```

Comment:
I'd remove it since one can always bypass a setter, and we don't want to be
in the business of preferentially error checking certain attributes at XML
write time.
On Jan 18, 2016 7:08 PM, "Sterling Harper" <notifications@github.com> wrote:

> In openmc/universe.py
> <https://github.com/mit-crpg/openmc/pull/563#discussion_r50058938>:
>
> > @@ -402,22 +402,23 @@ def create_xml_subelement(self, xml_element):
> >          if len(self._name) > 0:
> >              element.set("name", str(self._name))
> >
> > -        if isinstance(self._fill, openmc.Material):
> > -            element.set("material", str(self._fill._id))
> > +        if isinstance(self.fill, basestring):
> > +            assert self.fill.strip().lower() == 'void'
>
> The value is checked in the setter. I just thought I'd add this extra line
> of defense in case one of us accidentally bypasses the setter. If you don't
> like it I'll remove it without complaint
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mit-crpg/openmc/pull/563/files#r50058938>.
>